Dockerfile

```diff
@@ -22,8 +22,10 @@ ENV TF_VERSION="1.6.6" \
 COPY LICENSE /licenses/LICENSE
 
 # Install python
-RUN microdnf install -y python3.11 && \
-    update-alternatives --install /usr/bin/python3 python /usr/bin/python3.11 1
+RUN microdnf install -y python3.12 && \
+    update-alternatives --install /usr/bin/python3 python /usr/bin/python3.12 1 && \
```

Comment:
is python 3.12 required for provider sync or just another dependency update?

Reply:
Just a regular dependency update. As the sub-project must add the cdktf python packages to the `pyproject.toml`, it can also bump the python version.